sopel/bot.py

```diff
@@ -594,80 +737,139 @@ def register_urls(self, urls: Iterable) -> None:
 
     # message dispatch
 
+    @contextlib.contextmanager
+    def sopel_wrapper(
+        self,
+        trigger: Trigger,
+        rule: plugin_rules.AbstractRule,
+    ) -> Generator[Sopel, None, None]:
+        """Context manager to bind and unbind the bot to a trigger and a rule.
```

Comment:
Oh yes! I should have added a "TODO: find a good name", because I don't like it either! Something like `bind_context` or `wraps_context` would probably be better I guess? Or maybe `bind_response`? I used `sopel_wrapper` for exactly the reason you talk about (it matches `SopelWrapper`), and I also think it doesn't properly communicate what we want to achieve with that.

Reply:
Other names:

* `wraps_message`
* `wraps_trigger_and_rule`
* just `wraps` (because why not?)